ArchWiki/ArchWiki.py

```diff
@@ -168,15 +176,21 @@ def detect_language(self, title):
         match = re.match("^(.+?)([ _]\(([^\(]+)\))?$", title);
         if match:
             lang = match.group(3)
-            if lang in language_names:
+            if lang in self._language_names:
                 detected_language = lang
                 pure_title = match.group(1)
+            else:
+                detected_language = None
         return pure_title, detected_language
 
     def get_local_filename(self, title, basepath):
         """ Return file name where the given page should be stored, relative to 'basepath'.
         """
         title, lang = self.detect_language(title)
+
+        if lang is None:
+            return ''
```

Comment:
Can we just return `None` here?

Reply:
Yes, the output of get_local_filename is not added to any string.